msal/tests/Test.MSAL.NET.Unit/RequestsTests/DeviceCodeRequestTests.cs

```diff
@@ -0,0 +1,198 @@
+                 "\"device_code\":\"BAQABAAEAAADXzZ3ifr-GRbDT45zNSEFEfU4P-bZYS1vkvv8xiXdb1_zX2xAcdcfEoei1o-t9-zTB9sWyTcddFEWahP1FJJJ_YVA1zvPM2sV56d_8O5G23ti5uu0nRbIsniczabYYEr-2ZsbgRO62oZjKlB1zF3EkuORg2QhMOjtsk-KP0aw8_iAA\"," +
+                 "\"verification_url\":\"https://microsoft.com/devicelogin\"," +
+                 "\"expires_in\":\"900\"," +
+                 "\"interval\":\"5\"," +
+                 "\"message\":\"To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code B6SUYU5PL to authenticate.\"}";
+
+            return MockHelpers.CreateSuccessResponseMessage(SuccessMessage);
+        }
+
+        [TestMethod]
+        [TestCategory("DeviceCodeRequestTests")]
+        public void TestDeviceCodeAuthSuccess()
+        {
+            var parameters = CreateAuthenticationParametersAndSetupMocks(out HashSet<string> expectedScopes);
+
+            // Check that cache is empty
+            Assert.AreEqual(0, _cache.tokenCacheAccessor.AccessTokenCacheDictionary.Count);
+            Assert.AreEqual(0, _cache.tokenCacheAccessor.AccountCacheDictionary.Count);
+            Assert.AreEqual(0, _cache.tokenCacheAccessor.IdTokenCacheDictionary.Count);
+            Assert.AreEqual(0, _cache.tokenCacheAccessor.RefreshTokenCacheDictionary.Count);
+
+            DeviceCodeResult actualDeviceCodeResult = null;
+            DeviceCodeRequest request = new DeviceCodeRequest(parameters, result => 
+            {
+                actualDeviceCodeResult = result;
+                return Task.FromResult(0);
+            });
+            var task = request.RunAsync(CancellationToken.None);
+            task.Wait();
+            var authenticationResult = task.Result;
+            Assert.IsNotNull(authenticationResult);
+            Assert.IsNotNull(actualDeviceCodeResult);
+
+            Assert.AreEqual("client_id", actualDeviceCodeResult.ClientId);
+            Assert.AreEqual("BAQABAAEAAADXzZ3ifr-GRbDT45zNSEFEfU4P-bZYS1vkvv8xiXdb1_zX2xAcdcfEoei1o-t9-zTB9sWyTcddFEWahP1FJJJ_YVA1zvPM2sV56d_8O5G23ti5uu0nRbIsniczabYYEr-2ZsbgRO62oZjKlB1zF3EkuORg2QhMOjtsk-KP0aw8_iAA", actualDeviceCodeResult.DeviceCode);
+            Assert.AreEqual(5, actualDeviceCodeResult.Interval);
+            Assert.AreEqual("To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code B6SUYU5PL to authenticate.", actualDeviceCodeResult.Message);
+            Assert.AreEqual("B6SUYU5PL", actualDeviceCodeResult.UserCode);
+            Assert.AreEqual("https://microsoft.com/devicelogin", actualDeviceCodeResult.VerificationUrl);
+
+            CoreAssert.AreScopesEqual(expectedScopes.AsSingleString(), actualDeviceCodeResult.Scopes.AsSingleString());
+
+            // Validate that entries were added to cache
+            Assert.AreEqual(1, _cache.tokenCacheAccessor.AccessTokenCacheDictionary.Count);
+            Assert.AreEqual(1, _cache.tokenCacheAccessor.AccountCacheDictionary.Count);

```

Comment:
>AccountCacheDictionary [](start = 57, length = 22)

Which account will this be? We only have the client id,

Reply:
should be yes.  will validate/automate this with end to end tests as we discussed in person.